### DISCUSSION TITLE
JMX credentials (for Active MQ)

### DISCUSSION THREAD

**Discussion by StefanM66** [jmstoolbox/jmstoolbox#125] [Q&A] [answered]
Using ActiveMQ

Is there a possibility to set a separate JMX user/password?

I tried plan B to deactive JMX but then I ran into the issue
jms.watchTopicAdvisories=false, the property to the broker seems to stop working.

MQ Broker without JMX security
JMX true, jms.watchTopicAdvisories=false  -> PASSED
JMX false,jms.watchTopicAdvisories=false  -> FAILED  (no rights to access Advisories)

Looks like JMX credentials are the same from the session but these are the credentials to consume topic / read queues, not for JMX.

**Comment by titou10titou10** [comment, marked as answer]
The userid and password are indeed currently shared between JMX authentication and session authentication
It should not be a lot of work to have a separate identity for JMX
Created feature request #126

**Reply by StefanM66** [reply]
Wow, just noticed you added it at once.
I can be the tester :-)

**Comment by titou10titou10** [comment]
I just commit the change. see #126

**Comment by titou10titou10** [comment]
No I don't know when I will release v5.13.0.. Are you OK by building JTB from source? high level instructions are here: https://github.com/jmstoolbox/jmstoolbox/blob/dev/org.titou10.jtb.core/doc/buid_instructions.txt

**Reply by StefanM66** [reply]
Hi,

very fast. I followed your very good Eclipse instruction and got it running as newbie

Only noticed a problem when starting the application.

![jmstoolbox-application](https://user-images.githubusercontent.com/18614115/146394938-0b3a46c4-266b-480e-af2f-2e7cc3145203.PNG)

**Comment by titou10titou10** [comment]
There were a few eclipse plugin dependencies missing in the "product" definition due to eclipse version changes. it's fixed
Thanks for reporting.
